File: solutions/CF_N1_2024_Mistura.cpp
#include <bits/stdc++.h>

using namespace std;
using ll = long long;

auto solve(int N, size_t K, const vector<int>& as)
{
    map<int, int> hist;
    ll ans = 0;
    int R = 0;

    for (int L = 0; L < N; ++L)
    {
        while (R < N && hist.size() < K)
        {
            hist[as[R]]++;
            R++;
        } 

        if (hist.size() == K)
            ans += (N - R) + 1;

        hist[as[L]]--;

        if (hist[as[L]] == 0)
            hist.erase(as[L]);
    }

    return ans;
}


int main()
{
    ios::sync_with_stdio(false);

    int N, K;
    cin >> N >> K;

    vector<int> as(N);

    for (auto& a : as)
        cin >> a;

    cout << solve(N, K, as) << '\n';

    return 0;
}

File: solutions/CF_N1_2024_Mistura.py
N, K = map(int, input().split())
As = list(map(int, input().split()))

hist = {}
ans = 0
R = 0

for L in range(N):
    while R < N and len(hist) < K:

        if As[R] in hist:
            hist[As[R]] += 1
        else:
            hist[As[R]] = 1

        R += 1

    if len(hist) == K:
        ans += (N - R) + 1;

    hist[As[L]] -= 1

    if hist[As[L]] == 0:
        del hist[As[L]]

print(ans)
